### PR TITLE
Legacy event templates and docs

### PR DESCRIPTION
**Pull request by jskeet**
(A fairly boring set of commits...)

### CODE REVIEW

File: examples/Google.Cloud.Functions.Examples.FSharpLegacyEventFunction/Function.fs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/// a FirestoreEvent.)
+/// </summary>
+type Function() =
+    interface ILegacyEventFunction<StorageObject> with
+        /// <summary>
+        /// Logic for your function goes here. Note that a Legacy Event function just consumes an event;
+        /// it doesn't provide any response.
+        /// </summary>
+        /// <param name="payload">The Cloud Event your function should respond to.</param>
+        /// <param name="context">The Cloud Event your function should respond to.</param>
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.FSharpLegacyEventFunction/Function.fs:34]: I think the docs for the params are wrong.

File: examples/Google.Cloud.Functions.Examples.VbLegacyEventFunction/CloudFunction.vb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+''' a FirestoreEvent.)
+''' </summary>
+Public Class CloudFunction
+    Implements ILegacyEventFunction(Of StorageObject)
+
+    ''' <summary>
+    ''' Logic for your function goes here. Note that a Legacy Event function just consumes an event;
+    ''' it doesn't provide any response.
+    ''' </summary>
+    ''' <param name="payload">The Cloud Event your function should respond to.</param>
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.VbLegacyEventFunction/CloudFunction.vb:33]: And same here. It's probably wrong in the templates.

File: examples/Google.Cloud.Functions.Examples.VbLegacyEventFunction/CloudFunction.vb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    ''' <param name="context">The Cloud Event your function should respond to.</param>
+    ''' <param name="cancellationToken">A cancellation token that Is notified if the request Is aborted.</param>
```
Comment (amanda-tarafa) [inline examples/Google.Cloud.Functions.Examples.VbLegacyEventFunction/CloudFunction.vb:35]: nit: the automatic capitalization of reserved words: Is, twice. (that is awful)

**Comment by jskeet** [comment]
Thanks for all of these - will fix those parameter and capitalization issues, then merge on green.

**Pull request merged**